Tests/GoogleAITests/GenerativeModelTests.swift

```diff
@@ -568,114 +566,67 @@ final class GenerativeModelTests: XCTestCase {
-        let text = try XCTUnwrap(content.text)
-        textResponses.append(text)
+        responseCount += 1
       }
-    } catch {
-      errorResponse = error
+    } catch let GenerateContentError.internalError(rpcError as RPCError) {
+      XCTAssertEqual(rpcError.httpResponseCode, 499)
+      XCTAssertEqual(rpcError.status, .cancelled)
+
+      // Check the content count is correct.
+      XCTAssertEqual(responseCount, 2)
+      return
     }
 
-    // TODO: Add assertions for response content
-    XCTAssertEqual(textResponses.count, 2)
-    XCTAssertNotNil(errorResponse)
+    XCTFail("Expected an internalError with an RPCError.")
   }
 
   func testGenerateContentStream_nonHTTPResponse() async throws {
     MockURLProtocol.requestHandler = try nonHTTPRequestHandler()
 
-    let stream = model.generateContentStream("What sorts of questions can I ask you?")
-    var responseError: Error?
+    let stream = model.generateContentStream("Hi")
     do {
       for try await content in stream {
         XCTFail("Unexpected content in stream: \(content)")
       }
-    } catch {
-      responseError = error
-    }
-
-    XCTAssertNotNil(responseError)
-    let generateContentError = try XCTUnwrap(responseError as? GenerateContentError)
-    guard case let .internalError(underlyingError) = generateContentError else {
-      XCTFail("Not an internal error: \(generateContentError)")
+    } catch let GenerateContentError.internalError(underlying) {
+      XCTAssertEqual(underlying.localizedDescription, "Response was not an HTTP response.")
       return
     }
-    XCTAssertEqual(underlyingError.localizedDescription, "Response was not an HTTP response.")
+
+    XCTFail("Expected an internal error.")
   }
 
   func testGenerateContentStream_invalidResponse() async throws {
     MockURLProtocol
       .requestHandler = try httpRequestHandler(
-        forResource: "InvalidStreamingResponse",
-        withExtension: "json"
+        forResource: "streaming-failure-invalid-json",
+        withExtension: "txt"
       )
 
     let stream = model.generateContentStream(testPrompt)
-    var responseError: Error?
     do {
       for try await content in stream {
         XCTFail("Unexpected content in stream: \(content)")
       }
-    } catch {
-      responseError = error
-    }
-
-    XCTAssertNotNil(responseError)
-    let generateContentError = try XCTUnwrap(responseError as? GenerateContentError)
-    guard case let .internalError(underlyingError) = generateContentError else {
-      XCTFail("Not an internal error: \(generateContentError)")
-      return
-    }
-    let decodingError = try XCTUnwrap(underlyingError as? DecodingError)
-    guard case let .dataCorrupted(context) = decodingError else {
-      XCTFail("Not a data corrupted error: \(decodingError)")
-      return
-    }
-    XCTAssert(context.debugDescription.hasPrefix("Failed to decode GenerateContentResponse"))
-  }
-
-  func testGenerateContentStream_emptyContent() async throws {
```

Comment:
Isn't an empty content test useful?

Reply:
Definitely, it was a duplicate. It still exists here with the updated naming format https://github.com/google/generative-ai-swift/blob/a2e0eaf82c38eccb95ba4541297c5248574b2db8/Tests/GoogleAITests/GenerativeModelTests.swift#L408-L413